src/main/java/org/cbio/gdcpipeline/tasklet/FileMappingTasklet.java

```diff
@@ -0,0 +1,281 @@
+        JsonObject content = new JsonObject();
+        JsonArray values = new JsonArray();
+
+        for (String id : uuidToFilesMap.keySet()) {
+            values.add(id);
+        }
+
+        content.addProperty("field", "cases.case_id");
+        content.add("value", values);
+
+        filters.addProperty("op", "in");
+        filters.add("content", content);
+
+        node.add("filters", filters);
+        node.addProperty("format", "JSON");
+        node.addProperty("fields", "file_name,cases.case_id");
+
+        return node.toString();
+
+    }
+
+    protected ResponseEntity<String> callGdcApi(String url, String payload) {
+        if (uuidToFilesMap.isEmpty()) {
+            return null;
+        }
+        LOG.info(" PAYLOAD = " + payload);
+        LOG.info("Calling GDC API ");
+
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<String> entity = new HttpEntity<String>(payload.toString(), httpHeaders);
+        ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.POST, entity, String.class);
+
+        return response;
+
+    }
+
+    /* Response Format
+    {
+        "data": {
+            "hits": [
+               { "file_name": "XYZ.vcf.gz"
+                 "cases": [{"case_id": "ABC123"}]
+               },
+               { .... }
+            ]
+            "pagination":{...}
+        }
+    }
+
+
+     */
+
+    protected GdcApiResponse parseJsonResponse(ResponseEntity<String> response) {
+        LOG.info("Parsing API Response ");
+        Gson g = new Gson();
+        GdcApiResponse r = g.fromJson(response.getBody().toString(), GdcApiResponse.class);
+        return r;
+    }
+
+    protected String getCaseIdFromResponse(Hits hit) {
+        List<Case> caseList = hit.getCases();
+        if (caseList.size() == 1) {
+            // The only case_id
+            return caseList.get(0).getCase_id();
+        } else {
+            for (Case c : caseList) {
+                if (uuidToFilesMap.containsKey(c.getCase_id())) {
+                    return c.getCase_id();
+                }
+            }
+
+        }
+        return null;
+    }
+
+    protected void addDataToMap(GdcApiResponse resp) {
+        //Map : case_id --> file_name
+
+        for (Hits hit : resp.getData().getHits()) {
+            //Response is jumbled and can have a file name associated with many unrelated case_id's.
+            // E.g  MAF files. Identify the id sent in request
+            String case_id = getCaseIdFromResponse(hit);
+            uuidToFilesMap.get(case_id).add(hit.getFile_name());
+
+        }
+    }
+
+    protected RepeatStatus gdcApiRequest(String payload) {
+
+        int callCount = 0;
+        int totalCallCount = 1;
+        int from = 1;
+
+        while (callCount != totalCallCount) {
+            callCount += 1;
+            String url = GDC_API_ENDPOINT + "?from=" + from + "&size=" + MAX_RESPONSE_SIZE;
```

Comment:
Due to maf files the response size is very large for even a few inputs. Now since we ignore case_id's for maf files I think it should be ok while parsing response. But will add a TODO to do performance check.